### DISCUSSION TITLE
订阅关系一致与蓝绿部署

### DISCUSSION THREAD

**Discussion by bugCats** [apache/rocketmq#6837] [Q&A] [answered]
线上使用**蓝绿部署**模式发布新版本，但是rmq又需要**订阅关系一致**。

假如新版本**增加**、**修改**了topic、tag订阅，在蓝绿切换、以及共存阶段，对**订阅关系一致**有影响吗？有多大的影响？

**Comment by fuyou001** [comment, marked as answer]
如果使用了tag订阅，且在发布中订阅关系不一致（tag变化了），最严重可能导致部分消息收不到。
其它情况可能会导致消息重复，应用做好幂等就没什么问题。

**Reply by chenzlalvin** [reply]
最新的版本，更换topic、新增topic订阅不影响一致性判断，但更新过滤表达式，确实还是有影响，不能很好支持灰度。建议一般需要停掉旧的订阅，全部更换上新的表达式才能保证完整。

**Comment by flystar32** [comment]
@bugCats 有没有考虑过，通过修改 consumer  group 的方式来实现

**Reply by bugCats** [reply]
具体要怎么操作？

我想想，新实例使用新的consumerGroup，和旧实例不属于同一个消费者组。
那么同一个消息会被旧实例、和新实例都可以消费。如果保证幂等，好像也行？

但是新增消费者组，需要手动新增 `%RETRY%xxx`topic，多建几个，然后滚动使用，问题也不大？

好像可行？
